### ISSUE TITLE
Dynamic IP tuya devices lost on router restart 

### ISSUE THREAD

**Issue by Incongruent** [rospogrigio/localtuya#1113]
## Environment

- Last working localtuya version (if known and relevant): 
- Home Assistant Core version:  
- [] Are you using the Home Assistant Tuya Cloud component ? 
- [] Are you using the Tuya App in parallel ? 

## Steps to reproduce

1. Connect tuya devices to router using using DHCP auto assigned IP
2. Successful connect localtuya to devices in HA and confirm on/off works.
2. Reboot routers and access points
3. Confirm in HA UI that the tuya devices no longer respond to on/off actions and listed as Unavailable.
4. Restart HA
5. Confirm tuya devices are now available and can be turned on/off

## Configuration `configuration.yaml` or `config_flow`

```yaml

```

## Provide Home Assistant taceback/logs

```txt

```

**Comment by kuradi** [comment]
actually i have same problem lately. i noticed it after latest ha update

**Comment by donburch888** [comment]
Actually that is the correct behavior of DHCP, and nothing to do with HA.  

Your router's DHCP allocates short-term IP addresses. Normally when a DHCP lease expires (often after 24 hours) the device and router renegotiate, and almost always just renew the lease for another period - so as long as the router stays up the DHCP addresses behave almost as though they were static.

But when you reboot the router you clear all its internal tables - including its list of who it allocated each DHCP address to.  HA is unable to send messages to the devices because the router now has no record of which device was at which DHCP IP address. 
The device won't initiate a DHCP renegotiation until its lease expires - by which time that temporary IP address may have been allocated to another device.  It's like loosing your mobile phone without a backup.  You can replace the phone hardware, but can't contact your friends until you are able to ask their contact information ... often not until they ring or txt you next. 

When you restart HA, HA initiates a search for all devices on the network to find out their IP addresses, and this effectively rebuilds the routers internal DHCP table.

So, if you need to reboot your router regularly then you should either:
(a) automate a reboot of HA shortly afterward, to re-establish all the links to devices, or 
(b) set the manual DHCP entry (it goes by different names; my Draytek router calls it "Bind IP to MAC") in your router. This is the method I use for my IoT and network infrastructure devices.

**Comment by Beat-YT** [comment]
yes but there should be a way to refresh the IP address from tuya cloud...
